app/models/draw.rb

```diff
@@ -17,6 +17,8 @@ class Draw < ApplicationRecord
   validates :name, presence: true
   validates :status, presence: true
 
+  validate :cannot_lock_intent_if_undeclared
```

Comment:
So.... this runs all the time, not just when `lock_intent` is being set to true. I'm not sure why the tests didn't pick up on this, but it's causing 500 errors when I try to activate a draw with seeded data.

The reason is because `calculate_oversub_metrics` exits in the original request (since the draw is a draft so `policy(@draw).oversub_report?` is not permitted), but then somehow the change to the draw status persists even though DrawActivator.activate returns the failure case. This causes `show` to render, but now that oversub report policy returns true and the view errors since the variables weren't prepared.

Whew.

Adding `if: ->() { lock_intent_changed? }` fixes the issue (and I've confirmed that it still runs when you try to lock intent).